File: scala/src/main/scala/com/spoonacular/client/model/InlineResponse20028.scala
package com.spoonacular.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import spoonacular._
import com.spoonacular.client.model.InlineResponse20028Ingredients
import com.spoonacular.client.model.InlineResponse20028Nutrition
import com.spoonacular.client.model.InlineResponse20028Servings
import java.math.BigDecimal
import scala.collection.immutable.Seq

/**
 * 
 * @param id 
 * @param title 
 * @param badges 
 * @param importantBadges 
 * @param breadcrumbs 
 * @param generatedText 
 * @param imageType 
 * @param ingredientCount 
 * @param ingredientList 
 * @param ingredients 
 * @param likes 
 * @param nutrition 
 * @param price 
 * @param servings 
 * @param spoonacularScore 
 */
case class InlineResponse20028(id: Int,
                title: String,
                badges: Seq[String],
                importantBadges: Seq[String],
                breadcrumbs: Seq[String],
                generatedText: String,
                imageType: String,
                ingredientCount: Option[Int],
                ingredientList: String,
                ingredients: Seq[InlineResponse20028Ingredients],
                likes: BigDecimal,
                nutrition: InlineResponse20028Nutrition,
                price: BigDecimal,
                servings: InlineResponse20028Servings,
                spoonacularScore: BigDecimal
                )

object InlineResponse20028 {
    /**
     * Creates the codec for converting InlineResponse20028 from and to JSON.
     */
    implicit val decoder: Decoder[InlineResponse20028] = deriveDecoder
    implicit val encoder: ObjectEncoder[InlineResponse20028] = deriveEncoder
}

File: clojure/src/spoonacular_api/specs/inline_response_200_28.clj
(ns spoonacular-api.specs.inline-response-200-28
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [spoonacular-api.specs.inline-response-200-28-ingredients :refer :all]
            [spoonacular-api.specs.inline-response-200-28-nutrition :refer :all]
            [spoonacular-api.specs.inline-response-200-28-servings :refer :all]
            )
  (:import (java.io File)))


(def inline-response-200-28-data
  {
   (ds/req :id) int?
   (ds/req :title) string?
   (ds/req :badges) (s/coll-of string?)
   (ds/req :importantBadges) (s/coll-of string?)
   (ds/req :breadcrumbs) (s/coll-of string?)
   (ds/req :generatedText) string?
   (ds/req :imageType) string?
   (ds/opt :ingredientCount) int?
   (ds/req :ingredientList) string?
   (ds/req :ingredients) (s/coll-of inline-response-200-28-ingredients-spec)
   (ds/req :likes) float?
   (ds/req :nutrition) inline-response-200-28-nutrition-spec
   (ds/req :price) float?
   (ds/req :servings) inline-response-200-28-servings-spec
   (ds/req :spoonacularScore) float?
   })

(def inline-response-200-28-spec
  (ds/spec
    {:name ::inline-response-200-28
     :spec inline-response-200-28-data}))
